lib/dal.js

```diff
@@ -2,35 +2,60 @@ var Q = require('q');
 var fetchJSON = require('./fetchJSON.js');
 var customError = require('./customError.js');
 
+/**                                                                              
+ * Represents an error that has both a name and message.
+ * @param {String} name - the name/type of the error
+ * @param {String} message - the corresponding message
+ * Mainly used to avoid crashing the program, as it does with regular errors.
+ */
+function CustomError(name, message) {
+  this.name = name;
+  this.message = message;
+}
+
 /** 
  * Attempts to access configuration file data for the shared extension
+ *   If this is the first time accessing this said extension, then it creates
+ *   a new sharedExtension "object" and its relevant underlying data structures,
+ *   adds it to the array of extsInUse and returns it.
+ *   Simply returns the object in extsInUse if already previously created. This
+ *   is done so that the busy data of the trunk will be used when 
+ *   getSharedExtension is called by another user when a call is in session.
  * @param {string} confFile - the path and filename to the configuration file
  * @param {string} name - the name of the sharedExtension to access
  * @return {Q} - Q promise object
  */
 
 var getSharedExtension = function(confFile, name) {
-  var defer = Q.defer();
   var sharedExtension = {};
-
   sharedExtension.found = false;
+
   return fetchJSON(confFile).then(function(data) {
-    data.sharedExtensions.forEach(function(extension, i) {
+    for (var index = 0; index < data.sharedExtensions.length; index++){
+      var extension = data.sharedExtensions[index];
+
       if (extension[name]) {
```

Comment:
extension[name] is used to check if there is an extension with the name "name". You search through the data's sharedExtensions and check if they have a property "name". For example, a shared extension of name "42" would check if an extension in the array has a property of "42", and if so return that object. I do not think it would make sense to log an error for every extension that doesn't have that particular name, but I could be incorrect.